src/modifySongData.js

```diff
@@ -19,13 +33,21 @@ function calculateMean(vals) {
  * we do some smoothing between frames, based on our smooth factor
  * @param {number[]} energy - A list of energy values for a single frequency band
  *   (i.e. bass, mid, high) with values ranging from 0, 255
- * @param {number} numDeviations - How many standard deviations in either direction
- *   from the mean we want to include in our new range
- * @param {number} smoothFactor - What percentage of previous frames to average
- *   into next frame (should be a number between 0 and 1
+ * @param {options} object
+ * @param {number} options.numDeviations - How many standard deviations in either
+ *   direction from the mean we want to include in our new range
+ * @param {number} options.smoothFactor - What percentage of previous frames to
+ *   average into next frame (should be a number between 0 and 1
+ * @param {number[2]} representativeIndexRange - The min/max index in the range
+ *   of indices we want to look at when calculating our mean/std deviation
  */
-function getFrequencyEnergy(energy, numDeviations=1.5, smoothFactor=0.7) {
-  const nonZero = energy.filter(e => e > 0);
+function getFrequencyEnergy(energy, options = {}) {
+  const numDeviations = valueOrDefault(options.numDeviations, 1.5);
+  const smoothFactor = valueOrDefault(options.smoothFactor, 0.7);
+  const representativeIndexRange = valueOrDefault(options.representativeIndexRange ||
+    [0, energy.length]);
+
+  const nonZero = energy.slice(...representativeIndexRange).filter(e => e > 0);
   const mean = calculateMean(nonZero);
```

Comment:
Nit: Do you really mean `nonZero` here or `greaterThanZero`, or is it irrelevant because we never see negative values?

Reply:
It is irrelevant here because values are all positive.